pkg/registry/graph.go

```diff
@@ -0,0 +1,22 @@
+package registry
+
+type Package struct {
+	Name           string
+	DefaultChannel string
+	Channels       map[string]Channel
+}
+
+type Channel struct {
+	Head     BundleKey
+	Replaces map[BundleKey]map[BundleKey]struct{}
+}
+
+type BundleKey struct {
+	BundlePath string
+	Version    string //semver string
+	CsvName    string
+}
+
+func (b *BundleKey) IsEmpty() bool {
```

Comment:
Since we're using the `BundleKey` struct as a key in the map, should we add another method to check if two bundle keys are identical? From https://blog.golang.org/maps (`where KeyType may be any type that is comparable`) .

I guess we don't have to for the purposes of defining the map (Go does this for us) but it may be a helpful utility function to have alongside `b.IsEmpty()`. `b.IsEqual(a)`.

Reply:
I think that's a fair thing to suggest, but perhaps it should just be included when something that needs to do the comparison is written?